service/state/core.go

```diff
@@ -61,8 +61,13 @@ func NewCore(dir string) (*Core, error) {
 		return nil, fmt.Errorf("could not open database: %w", err)
 	}
 
+	return NewCoreFromDB(db)
+}
+
+func NewCoreFromDB(db *badger.DB) (*Core, error) {
+
```

Comment:
Why even keep the `NewCore` and not just make `NewCore` take a db at all times? As it stands, `NewCore` is untestable with unit tests unfortunately, despite having some interesting logic to test.

Reply:
Yeah, I don't mind it either way to be honest. If `NewCore` were to take in a db, we could also take care of the db `Close()` elsewhere (e.g. in main).

Additionally we could keep the alternative as `NewCoreFromPath` to serve as more of a helper function.